### PR TITLE
Helm flags for Global, Install and Upgrade helm commands

### PR DESCRIPTION
**Pull request by tjerkw**
This solves the following outstanding pull requests (they can all be closed):
#1284 #1506 #1507 #1445 #1499

It allows configuring flags to helm, for 

- Global helm flags (Similar to #1284)
- "helm install" flags
- "helm upgrade" flags

The code is similar to the code in kustomize, and kubectl. They all have the option to add flags to the deploy commands.

This also makes skaffold easier to use if helm comes with new flags, it will "just work". Since the flags are free text. Just add them, and run 👍 

Btw: The flags are set on the top level "_HelmDeploy_" struct. Not in the "_HelmRelease_" section in the yaml file.

This is because a "Global" flags list doesn't make sense per release. Then its not global anymore. 

Custom flags to "helm install/upgrade" do make sense per release. But most of the time you probably want to keep them similar. Thats why i didn't set the flag on the _HelmRelease_ struct.

Also: The flags clash a bit with the "_Wait_" and "_RecreatePods_" and "_Namespace_" fields in HelmRelease. However if you use both helm would complain and not run. So we're fine there.

### CODE REVIEW

File: examples/annotated-skaffold.yaml [hunk 1]
```diff
@@ -195,6 +195,13 @@ deploy:
     #   delete: [""]
 
  # helm:
+    # helm can be passed additional option flags either 
+    # on every command (Global), on install (Install) or on update (Update).
+    # Note: Parameters can also be set on Release level, but only some flags
+    # flags:
+    #   global: [""]
+    #   apply: [""]
```
Comment (priyawadhwa) [inline examples/annotated-skaffold.yaml]: I think this should be 'install' and delete should be 'upgrade'

File: integration/examples/annotated-skaffold.yaml [hunk 2]
```diff
@@ -194,7 +194,13 @@ deploy:
...
     #   delete: [""]
 
- # helm:
+ # helm:    
+    # helm can be passed additional option flags either 
+    # on every command (Global), on install (Install) or on update (Update).
+    # flags:
+    #   global: [""]
+    #   install: [""]
+    #   update: [""]
```
Comment (priyawadhwa) [inline integration/examples/annotated-skaffold.yaml]: should this be upgrade?

File: pkg/skaffold/schema/latest/config.go [hunk 3]
```diff
@@ -331,6 +331,18 @@ type KubectlFlags struct {
...
+	// Optional flags to send to the helm command
+	Flags HelmDeployFlags `yaml:"flags,omitempty"`
+}
+
+// HelmDeployFlags describes additional options flags that are passed on the command
+// line to helm either on every command (Global), on install (Install)
+// or on update (Update).
+type HelmDeployFlags struct {
+	Global  []string `yaml:"global,omitempty"`
+	Install []string `yaml:"apply,omitempty"`
```
Comment (priyawadhwa) [inline pkg/skaffold/schema/latest/config.go]: this should be 
```golang
Install []string `yaml:"install,omitempty"`
```

File: pkg/skaffold/schema/latest/config.go [hunk 4]
```diff
@@ -331,6 +331,18 @@ type KubectlFlags struct {
...
+	Update  []string `yaml:"delete,omitempty"`
```
Comment (priyawadhwa) [inline pkg/skaffold/schema/latest/config.go]: this should be

```golang
Upgrade []string `yaml:"upgrade,omitempty"`
```

File: pkg/skaffold/schema/latest/config.go [hunk 5]
```diff
@@ -331,6 +331,18 @@ type KubectlFlags struct {
...
+	Flags HelmDeployFlags `yaml:"flags,omitempty"`
+}
+
+// HelmDeployFlags describes additional options flags that are passed on the command
+// line to helm either on every command (Global), on install (Install)
+// or on update (Update).
+type HelmDeployFlags struct {
+	Global  []string `yaml:"global,omitempty"`
+	Install []string `yaml:"apply,omitempty"`
+	Upgrade []string `yaml:"delete,omitempty"`
```
Comment (priyawadhwa) [inline pkg/skaffold/schema/latest/config.go]: I think this should be
```golang
Upgrade []string `yaml:"upgrade,omitempty"`
```

File: integration/examples/annotated-skaffold.yaml [hunk 6]
```diff
@@ -194,7 +194,13 @@ deploy:
     #   apply: [""]
     #   delete: [""]
 
- # helm:
+ # helm:    
+    # helm can be passed additional option flags either 
+    # on every command (Global), on install (Install) or on update (Update).
```
Comment (priyawadhwa) [inline integration/examples/annotated-skaffold.yaml]: `update` should say `upgrade`

File: examples/annotated-skaffold.yaml [hunk 7]
```diff
@@ -195,6 +195,12 @@ deploy:
     #   delete: [""]
 
  # helm:
+    # helm can be passed additional option flags either 
+    # on every command (Global), on install (Install) or on update (Update).
```
Comment (priyawadhwa) [inline examples/annotated-skaffold.yaml]: `udpate` should say `upgrade`

File: integration/examples/annotated-skaffold.yaml (hunk 2 above)
Comment (tjerkw) [reply integration/examples/annotated-skaffold.yaml]: Yes should be updated now

File: examples/annotated-skaffold.yaml (hunk 7 above)
Comment (tjerkw) [reply examples/annotated-skaffold.yaml]: I updated all the update->upgrade mistakes!

File: integration/examples/annotated-skaffold.yaml (hunk 6 above)
Comment (tjerkw) [reply integration/examples/annotated-skaffold.yaml]: Fixed with https://github.com/GoogleContainerTools/skaffold/pull/1673/commits/52e919999c5485eb8ffda07e896df83970328b87

**Comment by tjerkw** [comment]
This is now ready to go. Thanks for the comments @priyawadhwa

File: pkg/skaffold/schema/latest/config.go (hunk 5 above)
Comment (priyawadhwa) [inline pkg/skaffold/schema/latest/config.go]: I think you forgot to change the yaml tag on this one

```golang
Upgrade []string `yaml:"upgrade,omitempty"`
```
:)

**Comment by tjerkw** [comment]
Is this ok to  go?

**Comment by priyawadhwa** [comment]
@tjerkw I think you just need to rebase, and then I can merge this!

**Comment by priyawadhwa** [comment]
Looks like travis failed -- 
```
--- FAIL: TestSchemas (0.26s)
	main_test.go:30: json schema files are not up to date.
           Please run `make generate-schemas` and commit the changes.
```
If you run that command and commit I think that should fix it!

**Comment by tjerkw** [comment]
Done. Forgot about it.

**Pull request merged**